### ISSUE TITLE
Check for parameter assignment notation in unset's argument

### ISSUE THREAD

**Issue by Lin-Buo-Ren** [koalaman/shellcheck#1550]
Currently, ShellCheck doesn't warn about parameter assignment expressions in the `unset` command's arguments.  Usually, this occurs when one assigned the parameters in a declare/local/export statement and copies their arguments to the unset command verbatim.

These mistakes will trigger <code>unset: `parameter=value': not a valid identifier</code> error during execution.

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:
```bash
#!/usr/bin/env bash
unset parameter=value
```

#### Here's what shellcheck currently says:
```
No issues detected!
```

#### Here's what I wanted or expected to see:
```
SCNNNN: Parameter assignment shouldn't occur in unset command, etc.
```

**Comment by koalaman** [comment]
This is definitely true.

I searched through ~10,000 scripts and could not find any instances of this though, nor of any other uses of `unset` with arguments that are not valid variable names, so while such a warning would have a low false positive rate, it does not appear to be common enough to warrant a warning. It's a good candidate if this changes in the future, so anyone finding this issue should reopen it. 

Thanks for the suggestion!
